ClosedXML/ClosedXML
# Allow using password for workbook protection.

#### What's this PR do?
Provide more methods to easily protect/unprotect workbook using password.
#### Where should the reviewer start?
Most work is in `ClosedXML/Excel/XLWorkbook.cs` and `ClosedXML/Excel/XLWorkbook_Save.cs`. 
#### How should this be manually tested?
Load up an excel and use `Workbook.Protect` or `Workbook.Unprotect` and then save and verify the protection of this workbook.
#### Any background context you want to provide?
The method `GetPasswordHash` in `ClosedXML/Excel/XLWorkbook.cs` is copied from `ClosedXML/Excel/XLSheetProtection.cs`.

#### Screenshots (if appropriate)
#### Questions:
- Is there a blog post?
- Does the knowledge base need an update?
- Does this add new (C#) dependencies?
